tests/test_iotedgedev_solution_tag.py

```diff
@@ -0,0 +1,85 @@
+import os
+import pytest
+from .utility import (
+    runner_invoke,
+)
+from iotedgedev.envvars import EnvVars
+from iotedgedev.output import Output
+from unittest import mock
+
+pytestmark = pytest.mark.e2e
+
+output = Output()
+envvars = EnvVars(output)
+test_solution_shared_lib_dir = os.path.join(os.getcwd(), "tests", "assets", "test_solution_shared_lib")
+
+
+# Test that cmd line tags (--tags) overrides DEVICE_TAGS from .env
+@ mock.patch.dict(os.environ, {"DEVICE_TAGS": "invalid_target"})
+def test_add_tags():
+    # Arrange
+    os.chdir(test_solution_shared_lib_dir)
+
+    # Act
+    result = runner_invoke(['tag', '--tags', '{"environment":"dev","building":"9"}'])
+
+    # Assert
+    assert 'TAG UPDATE COMPLETE' in result.output
+    assert '{"environment":"dev","building":"9"}' in result.output
+    assert 'ERROR' not in result.output
+
```

Comment:
should the test remove the tag at the end?

Reply:
Added bellow :)